subsys/debug/tracing/tracing_backend_usb.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2019 Intel corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <sys/util.h>
+#include <sys/atomic.h>
+#include <sys/byteorder.h>
+#include <usb/usb_device.h>
+#include <usb/usb_common.h>
+#include <usb_descriptor.h>
+#include <tracing_backend.h>
+#include <tracing_buffer.h>
+#include <debug/tracing_core.h>
+
+#define USB_TRANSFER_ONGOING               1
+#define USB_TRANSFER_FREE                  0
+
+#define TRACING_IF_IN_EP_ADDR              0x81
+#define TRACING_IF_OUT_EP_ADDR             0x01
+
+/* Max packet size for endpoints */
+#define BULK_EP_MPS                        32
+
+struct usb_device_desc {
+	struct usb_if_descriptor if0;
+	struct usb_ep_descriptor if0_in_ep;
+	struct usb_ep_descriptor if0_out_ep;
+} __packed;
+
+static atomic_t transfer_state;
```

Comment:
Since this is busy-polled between differing contexts, you may want `volatile`.

Reply:
Thanks